### PR TITLE
[GEOS-11455] Update developers guide with GSIP-220 security policy

### PR DESCRIPTION
**Pull request by jodygarnett**
[GEOS-11455](https://osgeo-org.atlassian.net/browse/GEOS-11455) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

While we updated SECURITY.md there the process from [GSIP-220](https://github.com/geoserver/geoserver/wiki/GSIP-220) was not yet added to the developers guide.

I have adapted the text of GSIP-220 as part of this pull request.
  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: doc/en/developer/source/policies/security.rst [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+Security Procedure
+==================
+
+This page covers some of the lines of communication and tools available to enact our public security policy (see `SECURITY.md <https://github.com/geoserver/geoserver/blob/main/SECURITY.md>`__).
+
+Vulnerability reporting and discussion
+--------------------------------------
+
+Our security policy asks community to take care in reporting and discussion of security vulnerabilities:
```
Comment (bradh) [inline doc/en/developer/source/policies/security.rst:11]:
```suggestion
Our security policy asks the community to take care in reporting and discussion of security vulnerabilities:
```

File: doc/en/developer/source/policies/security.rst [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+ **Reporting a Vulnerability**
+ 
+ If you encounter a security vulnerability in GeoServer please take care to report in a responsible fashion:
```
Comment (bradh) [inline doc/en/developer/source/policies/security.rst:15]:
```suggestion
 If you encounter a security vulnerability in GeoServer please take care to report it in a responsible fashion:
```

File: doc/en/developer/source/policies/security.rst [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  Recognize that such communication is outside our control and do not further engage. 
+  
+* Public CVE: Most reliable response is to locate the CVE on advisory database and issue a pull-request clarification.
+  
+  * https://github.com/advisories?query=geoserver
+  * https://github.com/advisories?query=geotools
+  
+  You may also go through the steps of "disputing" a CVE, this involves contacting the original numbering authority (each of which have their own procedures).
+  
+  Keepin in mind that security researcher may have obligation to report to their national numbering authority.
```
Comment (bradh) [inline doc/en/developer/source/policies/security.rst:59]:
```suggestion
  Keep in mind that security researchers may have obligations to report to their national numbering authority.
```

File: doc/en/developer/source/policies/security.rst [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+* This is a long game, so prioritize sustainability and our geoserver-security volunteers.
+* Recognize that the reporter is likely stressed, and may not have much flexibility due legal obligations around vulnerability disclosure.
+* This is a sustainability challenge, so do not be shy about seeking sponsorship.
+
+geoserver-security list participation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Participation in the geoserver-security, like commit access, is volunteer based and reflects trust.
+
+* To volunteer email geoserver-security@lists.osgeo.org stating your interest.
```
Comment (bradh) [inline doc/en/developer/source/policies/security.rst:74]:
```suggestion
* To volunteer, email geoserver-security@lists.osgeo.org stating your interest.
```

File: doc/en/developer/source/policies/security.rst [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+* If you are unable to participate send and email to step down.
```
Comment (bradh) [inline doc/en/developer/source/policies/security.rst:76]:
```suggestion
* If you are unable to participate send an email to step down.
```

File: doc/en/developer/source/policies/security.rst [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+
+  We understand that short term committments can get in the way of sustainability initiatives – you are welcome to rejoin at any time.
```
Comment (bradh) [inline doc/en/developer/source/policies/security.rst:78]:
```suggestion
  We understand that short term commitments can get in the way of sustainability initiatives – you are welcome to rejoin at any time.
```

File: doc/en/developer/source/policies/security.rst [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+  
+  We will also retire inactive members as needed
```
Comment (bradh) [inline doc/en/developer/source/policies/security.rst:80]:
```suggestion
  We will also retire inactive members as needed.
```

File: doc/en/developer/source/policies/security.rst [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+  5. The CVE vulnerability is published with mitigation and patch instructions
+
+  This represents a balance between transparency and participation that does not overwhelm participants. Those seeking greater visibility are encouraged to volunteer with the geoserver-security list; or work with one of the `commercial support providers <https://geoserver.org/support/>`__ who participate on behalf of their customers.
+
+Working with vulnerability reports
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. When working with a "triage" or "draft" vulnerability:
+
+   * Invite reported to participate, they are part of the team addressing the issue,
```
Comment (bradh) [inline doc/en/developer/source/policies/security.rst:121]:
```suggestion
   * Invite reporter to participate, they are part of the team addressing the issue,
```

File: doc/en/developer/source/policies/security.rst [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+   * Give credit to the reporter any anyone else involved using the allocated fields
```
Comment (bradh) [inline doc/en/developer/source/policies/security.rst:122]:
```suggestion
   * Give credit to the reporter and anyone else involved using the allocated fields
```

File: doc/en/developer/source/policies/security.rst [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+   * Do not immediately request a CVE, and be sure to confirm the vulnerability (for example with a proof of concept) first.
+   * A vulnerability flagged from a dependency scan does not automatically indicate that an exploit is available in GeoServer.
+   * For popular concerns (like spring-framework upgrade) add each reporter to same advisory.
+   
+   Be cautious that a security research may only be tasked with reporting the issue, and be unavailable
```
Comment (bradh) [inline doc/en/developer/source/policies/security.rst:127]:
```suggestion
   Be cautious that a security researcher may only be tasked with reporting the issue, and be unavailable
```

File: doc/en/developer/source/policies/security.rst [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+   * Package: Always report `org.geoserver.web:gs-web-app` as `geoserver.war` as a useful way to document that the `geoserver.war` includes other jars
+   * Affected versions: It is difficult to communicate version ranges, due to limitations in CVE advistory processing, requiring multiple lines.
+   
+   .. figure:: img/cve-version-range.png
+      
+      Package guidance and version range
+
+3. Work on providing a fix, mitigation instructions, or best-practice clarification for documentation.
+   
+   * The use of github private repository associated with a advistory should be used with caution.
```
Comment (bradh) [inline doc/en/developer/source/policies/security.rst:141]:
```suggestion
   * The use of github private repository associated with an advisory should be used with caution.
```

File: doc/en/developer/source/policies/security.rst [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+   
+     If making extensive code changes keep in mind that automations, including QA automations, are not available to be run as part of the pull-request review process.
+     
+     This may be appropriate when updating documentation / best practice information.
+     
+     They key advantage is several pull-requests can be managed at once, and merged at the same time as disclosure (see below).
+   
+   * If you wish to work on a public pull-request (to take advantage of workflow automations) take care that test-cases, commit messages, and documentation updates do not immediately reveal the vulnerability.
+   
+   * Although not ideal it is possible to resolve some security issues by documenting a best-practice in the production consideration of the user guide.
```
Comment (bradh) [inline doc/en/developer/source/policies/security.rst:151]:
```suggestion
   * Although not ideal, it is possible to resolve some security issues by documenting a best-practice in the production consideration of the user guide.
```

**Review by petersmythe — APPROVED** [review]

**Comment by jodygarnett** [comment]
Thanks, it is a start - we can adapt over time.

**Pull request merged**